index.html

```diff
@@ -14017,49 +14026,42 @@ <h4 id="PrimaryGlobal">4.3.6. [Global] and [PrimaryGlobal]</h4>
+                  <ol>
+                    <li>If <var>value</var>'s interface is a <a class="dfnref" href="#dfn-single-operation-callback-interface">single operation callback interface</a> and <a class="external" href="https://tc39.github.io/ecma262/#sec-iscallable">IsCallable</a>(<var>O</var>) is true, then set <var>X</var> to <var>O</var>.</li>
+                    <li>Otherwise, <var>opName</var> must be supplied. Set <var>X</var> to <a class="external" href="https://tc39.github.io/ecma262/#sec-get-o-p">Get</a>(<var>O</var>, <var>opName</var>).</li>
+                  </ol>
+                </li>
                 <li>If <a class="external" href="https://tc39.github.io/ecma262/#sec-iscallable">IsCallable</a>(<var>X</var>) is false, then <a href="#ecmascript-throw" class="dfnref external">throw a <span class="estype">TypeError</span></a> exception.</li>
-                <li>Let <var>this</var> be <var>O</var> if the interface is not a <a class="dfnref" href="#dfn-single-operation-callback-interface">single operation callback interface</a>
-                  or if <a class="external" href="https://tc39.github.io/ecma262/#sec-iscallable">IsCallable</a>(<var>O</var>) is false,
-                  and the <a class="dfnref" href="#dfn-callback-this-value">callback this value</a> otherwise.</li>
+                <li>If <var>value</var>'s interface is not a <a class="dfnref" href="#dfn-single-operation-callback-interface">single operation callback interface</a>, or if <a class="external" href="https://tc39.github.io/ecma262/#sec-iscallable">IsCallable</a>(<var>O</var>) is false, set <var>thisArg</var> to <var>O</var> (overriding the provided value).</li>
                 <li>
                   Let <var>arg</var><sub>0..<var>n</var>−1</sub> be a list of
                   ECMAScript values, where <var>arg</var><sub><var>i</var></sub> is the result
                   of <a class="dfnref" href="#dfn-convert-idl-to-ecmascript-value">converting</a>
                   <var>idlarg</var><sub><var>i</var></sub> to an ECMAScript value.
                 </li>
-                <li>Let <var>script</var> be the <a class="dfnref" href="#dfn-callback-context">callback context</a> associated with <var>V</var>.</li>
-                <li>Push <var>script</var> on to the <a class="dfnref external" href="http://www.whatwg.org/specs/web-apps/current-work/multipage/webappapis.html">stack of incumbent scripts</a>. <a href="#ref-HTML">[HTML]</a></li>
+                <li>Let <var>execution context</var> be <a class="external" href="https://tc39.github.io/ecma262/#sec-getfunctionrealm">GetFunctionRealm</a>(<var>X</var>)'s <a class="dfnref external" href="https://html.spec.whatwg.org/multipage/webappapis.html#concept-realm-settings-object">settings object</a>'s <a class="external" href="https://html.spec.whatwg.org/multipage/webappapis.html#realm-execution-context">realm execution context</a>.</li>
```

Comment:
No, this doesn't work, I think.  The reason it doesn't work is that you're too late: you may already have thrown an exception from the `Get` above, and that needs to report the exception, and that needs the entry settings stuff, because that's where exceptions get reported, right?

I put together two interesting testcases: <http://web.mit.edu/bzbarsky/www/testcases/settings-objects/entry-settings-1.html> and <http://web.mit.edu/bzbarsky/www/testcases/settings-objects/entry-settings-2.html> -- on both you need to click the "click me" text .  The first one uses an element from window A, adds a listener that is an object created from window B, whose `handleEvent` property is a function from window C that throws.  The second one uses an element from window A, adds a listener that is an object created from window B, whose `handleEvent` property has a getter that is a function from window C that throws.

In Firefox, both cases use window B as the place to fire error events, and entry settings object. 

In Chrome, they both seem to use window A as the place to fire error events (because of where the listeners are attached, iirc how this works in Chrome); not sure whether that precisely corresponds to entry settings in Chrome.

In Safari, the behavior is the same as in Chrome (uses window A) for the first testcase and the second testcase doesn't fire any error listeners.  Again, not sure what the actual entry settings thing is.

In Edge, the first testcase doesn't fire error listeners afaict and the second one fires them in window B.   Again, not sure what the actual entry settings thing is.

This spec as written (with your changes) would fire listeners in window C on the first testcase and has undefined behavior on the second one, as far as I can tell.